### PR TITLE
Some features added

### PR DESCRIPTION
**Pull request by Dewstar**
Added Delete in CurlClient
Added function to resolve an issue in Api
Added function to reopen an issue in Api
Added function to delete watchers in Api
Added function to delete comments in Api
Added function to delete attachments in Api

### CODE REVIEW

**Comment by aik099** [comment]
Not sure if it's a good idea to hardcode any transition-specific actions in general API class. Since in JIRA workflow and transition names can be customized then for some users such call might end up with error.

Better to have universal transition method, that would accept:
* transition name
* fields to change

**Comment by aik099** [comment]
I've changed my mind. Please rebase this and it can be merged. We can later refactor to remove duplicate transition finding code code and make actual transition names customizable.

**Comment by aik099** [comment]
Please rebase and solve conflicts.

File: src/Jira/Api/Client/CurlClient.php [hunk 1]
```diff
@@ -82,6 +82,8 @@ public function sendRequest($method, $url, $data = array(), $endpoint, Authentic
             } else {
                 curl_setopt($curl, CURLOPT_POSTFIELDS, json_encode($data));
             }
+        } else if($method == "DELETE") {
```
Comment (aik099) [inline src/Jira/Api/Client/CurlClient.php]: This is already addressed in #76.

**Comment by jpastoor** [comment]
Planning on rebasing this myself if original author does not want to after #137 is finished.

**Pull request closed without merging**